util/model_initializer.py

```diff
@@ -80,11 +95,7 @@ def _create_path(self, matrix_name, matrix_shape, config=None):
         if 'n_iterations' not in config.keys():
             config['n_iterations'] = self.config['n_iterations']
         config['n_factors'] = matrix_shape[1]
-        generated_key = ''
         config['n_rows'] = n_rows
-        keys_array = sorted(config)
-        for key in keys_array:
-            generated_key += key + ':'
-            generated_key += str(config[key]) + ','
-        path = generated_key.strip(',') + matrix_name
-        return os.path.join(os.path.dirname(os.path.realpath(__file__)), '../', self.folder, path + '.dat')
+        path = self._generate_path(config, matrix_name)
+        base_dir = os.path.dirname(os.path.realpath(__file__))
+        return os.path.join(os.path.dirname(base_dir), self.folder, path + '.dat')
```

Comment:
It will, just with the minor change of using '-' instead ':' because the ':' can make some problems in bash later on if we want to use these files, and probably it will be problematic on different operating systems.
And I don't see the problem if a function is absolutely useless without the others, functions are usually dependent on each others.